apps/music/views/albums/view.js

```diff
@@ -10,66 +10,28 @@ var AlbumsView = View.extend(function AlbumsView() {
 
   this.client.on('databaseChange', () => this.update());
 
-  this.configureList();
-  this.configureSearch();
+  this.setupSearch();
+  this.setupList();
+
   this.update();
 });
 
-AlbumsView.prototype.configureList = function() {
-  // Scroll search out of view, even when
-  // there aren't enough list items to scroll.
-  this.list.scrollTop = this.searchBox.HEIGHT;
-  this.list.minScrollHeight = `calc(100% + ${this.searchBox.HEIGHT}px)`;
-
-  this.list.configure({
-    getItemImageSrc: (item) => this.getThumbnail(item.name)
-  });
-
-  // Show the view only when list has something
-  // rendered, this prevents Gecko painting unnecessarily.
-  this.once(this.list, 'rendered', () => document.body.hidden = false);
-};
-
-AlbumsView.prototype.configureSearch = function() {
-  this.searchBox.addEventListener('search', (evt) => this.search(evt.detail));
-
-  this.searchResults.addEventListener('open', () => {
-    this.client.method('searchOpen');
-  });
-
-  this.searchResults.addEventListener('close', () => {
-    this.client.method('searchClose');
-    this.list.scrollTop = this.searchBox.HEIGHT;
-  });
-
-  this.searchResults.addEventListener('resultclick', (evt) => {
-    var link = evt.detail;
-    if (link) {
-      this.client.method('navigate', link.getAttribute('href'));
-    }
-  });
-
-  this.searchResults.getItemImageSrc = (item) => this.getThumbnail(item.name);
-};
-
 AlbumsView.prototype.update = function() {
-  this.getAlbums().then((albums) => {
+  return this.getAlbums().then((albums) => {
     this.albums = albums;
-    this.render();
+    return this.render();
   });
 };
 
 AlbumsView.prototype.destroy = function() {
   this.client.destroy();
-
   View.prototype.destroy.call(this); // super(); // Always call *last*
 };
 
 AlbumsView.prototype.render = function() {
   View.prototype.render.call(this); // super();
-
-  this.list.model = this.albums;
-  this.list.cache();
+  return this.list.setModel(this.albums)
```

Comment:
IMO if functions are async, it's good convention to return a `Promise` to indicate when they're done.